extensions/azurePublishNew/src/components/shared/autoComplete/hooks/useSelection.ts

```diff
@@ -0,0 +1,33 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+/**
+ * Copyright (c) Microsoft. All rights reserved.
+ */
+
+import { ISelection, Selection, SelectionMode } from 'office-ui-fabric-react/lib/DetailsList';
+import React from 'react';
+
+/**
+ * A hook that can be used in conjunction with the table component. This provides
+ * a way of subscribing to the selected rows in the table. It returns an object
+ * containing the selection instance and an array containing the currently
+ * selected items.
+ *
+ * @param selectionMode An optional selection mode to use with the hook.
+ */
+export function useSelection<T>(selectionMode?: SelectionMode) {
```

Comment:
this would fail if a selection ref exists and one changes the selectionMode from outside

Reply:
I have changed the name to initialSelectionMode, that way it is clear to the caller, that it is only set for the first time.